### ISSUE TITLE
FastAPI framework for C/C++

### ISSUE THREAD

**Issue by ZigRazor** [open-source-ideas/ideas#315]
## Project description

With more and more C++ user and the advent of microservices, I think is important have a framework like [FastAPI](https://fastapi.tiangolo.com/) or [Flask](https://flask.palletsprojects.com/en/2.0.x/) also in C++ language. This kind of framework can open the doors to efficent database or web services(or microservices) based on C++ language.
The desiderable is to have a platform independent framework with simple API and high performance for building API for microservices or web applications.

This will grants:

- Faster Coding
- Fewer Bugs
- Robust application

If possible this framework should be complaint with [OpenAPI](https://github.com/OAI/OpenAPI-Specification) specification and [JSON Schema](https://json-schema.org/)

Some projects like [Crow](https://crowcpp.org/) try to do this, but this projects are old or no more maintained.

## Relevant Technology

- C/C++  Language
- [JSON Schema](https://json-schema.org/)
- [OpenAPI](https://github.com/OAI/OpenAPI-Specification) specification
- cross-platform library, but in a first moment it's possible to start from linux distribution like Fedora,Debian,Ubuntu,RedHat etc,etc,

### Complexity

- [ ] Beginner - This project requires no or little prior knowledge of the technolog(y|ies) specified to contribute to the project
- [x] Intermediate - The user should have some prior knowledge of the technolog(y|ies) to the point where they know how to use it, but not necessarily all the nooks and crannies of the technology
- [ ] Advanced - The project requires the user to have a good understanding of all components of the project to contribute

### Required time (ETA)

- [ ] Little work - A couple of days
- [ ] Medium work - A week or two
- [x] Much work - The project will take more than a couple of weeks and serious planning is required

### Categories

- [ ] Mobile app
- [ ] IoT
- [x] Web app
- [ ] Frontend/UI
- [ ] AI/ML
- [x] APIs/Backend
- [ ] Voice Assistant
- [x] Developer Tooling
- [ ] Extension/Plugin/Add-On
- [ ] Design/UX
- [ ] AR/VR
- [ ] Bots
- [ ] Security
- [ ] Blockchain
- [ ] Futuristic Tech/Something Unique

### Support

For this project I can give full support, as designer and as developer and also as project manager with my experience in open-source project and as C++ expert,, but I need some help in writing code, the amount of work is very very big.
Thank you in advance!

**Comment by leoferaderonugraha** [comment]
Is this still available?

**Comment by ZigRazor** [comment]
Yes, it is still available, do you have some experience on these framework or in C++? 
The project group is still under costrctution, do you want partecipate?

**Comment by leoferaderonugraha** [comment]
I've worked on some projects built with FastAPI and I'm somewhat familiar with C & C++.
How do I join the project group?

**Comment by ZigRazor** [comment]
At this moment the group is not formed yet, but if you want we can start a new group and open the project, and start working.
What do you think?

**Comment by yusufmalikul** [comment]
It would be great if we have a Slack channel somehow.

**Comment by leoferaderonugraha** [comment]
Sure, I'm up for it.

**Comment by ZigRazor** [comment]
@yusufmalikul do you want to be part of the organization?

**Comment by ZigRazor** [comment]
@pranavhari2 I invited also you!

**Comment by kmadridr** [comment]
Sounds great

**Comment by ZigRazor** [comment]
do you want partecipare @kmadridr ?

**Comment by kkgg0521** [comment]
我想参加 现在还有机会么

**Comment by Spidy104** [comment]
I want to commit to the project

**Comment by pmjfontes** [comment]
Hi, I have experience with FASTAPI and also C/C++. I would like to participate in this project.

**Comment by ZigRazor** [comment]
@pmjfontes we have to start soon, but we need to discuss some technical details.
I add you to the organization group

**Comment by ZigRazor** [comment]
@John98Zakaria I add you to the working group!

**Comment by Subhimukbel** [comment]
Please add me I want to participate

**Comment by anshulsahni** [comment]
Hi folks, I would like to participate in this project, if there is still some room ?

**Comment by ZigRazor** [comment]
yes, I add you to the group

**Comment by Spidy104** [comment]
> yes, I add you to the group

Hello can you also add me I would love to help

**Comment by ZigRazor** [comment]
YES!

**Comment by AbhishekSharma20** [comment]
Please add me as well. I have intermidete level experience with both python and cpp.

**Comment by tomycostantino** [comment]
I would love to participate, I just started with the idea of building something similar but I see that this is a great initiative.

**Comment by ZigRazor** [comment]
@tomycostantino I add you to the group, then if we want to start we can

**Comment by ZigRazor** [comment]
@tomycostantino I see you work, you can start from your code and commit it in https://github.com/HardSoftwareFoundation/CXXFastAPI

**Comment by tomycostantino** [comment]
@ZigRazor thanks mate. I've been building sort of a foundational web server so far, happy to share it and see where we go

**Comment by Lovening** [comment]
Hello, my friend. I have experience in C++ and network programming, but I am not sure how to contribute to this project. I would like to offer my humble help to the project.

**Comment by ZigRazor** [comment]
@Lovening  I add you to the working-group

**Comment by rajaParmar** [comment]
@ZigRazor, I like this idea .. would like to contribute :)

**Comment by ZigRazor** [comment]
I add you to the working group

**Comment by raustin9** [comment]
I also am an intermediate in C++ and have been using fastapi in python for university research. I am interested in contributing to this as well. I am still a student and working so my time might be limited but this is a project I would find incredibly useful, so I would love to contribute if possible!

**Comment by ZigRazor** [comment]
I add you to the group @raustin9

**Comment by deepwilson** [comment]
is it still on ?

**Comment by ZigRazor** [comment]
yes I can Add you to the group @deepwilson

**Comment by helauren42** [comment]
I would love to work on this, not sure I'm qualified but I have time on my side.

**Comment by ZigRazor** [comment]
I add you to the group @helauren42

**Comment by mohammad-nazari** [comment]
Greate Idea

**Comment by YashArote** [comment]
I am trying to do something similar , FastApi style Cpp framework. You can check it out https://github.com/YashArote/fastapi-cpp

**Comment by ZigRazor** [comment]
@YashArote  what do you think about a possible transfer of the repo under the team (https://github.com/orgs/HardSoftwareFoundation/teams/cxxfastapi_team) of CXXFAST_API?

I can introduce you in my Foundations (https://github.com/HardSoftwareFoundation)

**Comment by YashArote** [comment]
Appreciate the invitation! I’m keeping things as-is for the moment, but I’ll definitely keep the org in mind as things evolve.

**Comment by ZigRazor** [comment]
Thank you so much @YashArote

**Comment by madhurbhutada** [comment]
Hi, @ZigRazor , I'd like to join the group too! I am interested in contributing to this. :)

**Comment by manit2004** [comment]
Hi @ZigRazor, Seems interesting ! Please add me to the group.
